service/ntfy/ntfy.go

```diff
@@ -0,0 +1,174 @@
+type Service struct {
+	client     *http.Client
+	serverURLs []string
+}
+
+func defaultHTTPClient() *http.Client {
+	return &http.Client{
+		Timeout: 5 * time.Second,
+	}
+}
+
+// DefaultServerURL is the default server to use for the Ntfy service.
+const DefaultServerURL = "https://ntfy.sh"
+
+// normalizeServerURL normalizes the server URL. It prefixes it with https:// if it's not already and appends a slash
+// if it's not already there. If the serverURL is empty, the DefaultServerURL is used. We're not validating the url here
+// on purpose, we leave that to the http client.
+func normalizeServerURL(serverURL string) string {
+	if serverURL == "" {
+		return DefaultServerURL
+	}
+
+	// Normalize the url
+	if !strings.HasPrefix(serverURL, "http") {
+		serverURL = "https://" + serverURL
+	}
+	if !strings.HasSuffix(serverURL, "/") {
+		serverURL = serverURL + "/"
+	}
+
+	return serverURL
+}
+
+// AddReceivers adds server URLs to the list of servers to use for sending messages.
+func (s *Service) AddReceivers(serverURLs ...string) {
+	for _, serverURL := range serverURLs {
+		serverURL = normalizeServerURL(serverURL)
+		s.serverURLs = append(s.serverURLs, serverURL)
+	}
+}
+
+// NewWithServers returns a new instance of Ntfy service. You can use this service to send messages to Ntfy. You can
+// specify the servers to send the messages to. By default, the service will use the default server
+// (https://api.day.app/) if you don't specify any servers.
+func NewWithServers(serverURLs ...string) *Service {
+	s := &Service{
+		client: defaultHTTPClient(),
+	}
+
+	if len(serverURLs) == 0 {
+		serverURLs = append(serverURLs, DefaultServerURL)
+	}
+
+	// Calling service.AddReceivers() instead of directly setting the serverURLs because we want to normalize the URLs.
+	s.AddReceivers(serverURLs...)
+
+	return s
+}
+
+// New returns a new instance of Ntfy service. You can use this service to send messages to Ntfy. By default, the
+// service will use the default server (https://ntfy.sh).
+func New() *Service {
+	return NewWithServers()
+}
+
+// postData is the data to send to the Ntfy server.
+type postData struct {
+	Topic    string   `json:"topic"`
+	Message  string   `json:"message,omitempty"`
+	Title    string   `json:"title,omitempty"`
+	Tags     []string `json:"tags,omitempty"`
+	Priority int      `json:"priority,omitempty"`
+	Attach   string   `json:"attach,omitempty"`
+	Filename string   `json:"filename,omitempty"`
+	Click    string   `json:"click,omitempty"`
+	Actions  []struct {
+		Action string `json:"action"`
+		Label  string `json:"label,omitempty"`
+		URL    string `json:"url"`
+	} `json:"actions,omitempty"`
+}
+
+func (s *Service) send(ctx context.Context, serverURL, topic, content string) (err error) {
+	if serverURL == "" {
+		return errors.New("server url is empty")
+	}
+
+	bodyByte := []byte(content)
+
+	isJson := json.Valid(bodyByte)
+	if !isJson {
+		err := errors.New("Invalid JSON Body")
+		return err
+	}
```

Comment:
Potential simplification.
```suggestion
	if !json.Valid(bodyByte) {
		return errors.New("Invalid JSON Body")
	}
```